File: ROS2_WS/src/my_python_pkg/my_python_pkg/Subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class SubscriberNode(Node):

    def __init__(self):
        super().__init__("Subscriber")
        self.subscriber_ = self.create_subscription(
            String, "publisher_node", self.callback_robot_news, 10)
        self.get_logger().info("Subscriber has started")

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)

    

def main(args=None):
        rclpy.init(args=args)
        node = SubscriberNode()
        rclpy.spin(node)
        rclpy.shutdown()

if __name__ == "__main__":
    main()

File: ROS2_WS/src/my_cpp_package/src/subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SubscriberNode : public rclcpp::Node{
    public:
    SubscriberNode() : Node("Subscriber_Node"){
        subscriber_ = this->create_subscription<example_interfaces::msg::String>("publisher_node", 10, std::bind(&SubscriberNode::callbackSubscriber, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Subscriber node has started");

    }
    private:
    void callbackSubscriber(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SubscriberNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}